### PR TITLE
Prevent "Blocked host:" message 

### PR DESCRIPTION
**Pull request by PromoFaux**
When using the latest container, I came up against the below issue, which was preventing Monzo webhooks from being processed.

image

Initially I just mounted the `development.rb` file and hardcoded `config.hosts << myhostname.com`, but that seemed really lazy, so figured I'd find a better way to do it that others may benefit from. 

I also have made some tweaks to the GHA config so that it can be pushed to ones own repository

Proof of the GHA config working: https://github.com/PromoFaux/fintech-to-ynab/runs/3776908590?check_suite_focus=true

(I've also no idea if `.nil?` is the correct way to do this, I stackoverflowed my way to it, not knowing Ruby)

### CODE REVIEW

File: config/environments/development.rb [hunk 1]
```diff
@@ -38,4 +38,10 @@
   # Use an evented file watcher to asynchronously detect changes in source code,
   # routes, locales, etc. This feature depends on the listen gem.
   config.file_watcher = ActiveSupport::EventedFileUpdateChecker
+
+  # Prevent "Blocked host: hostname"
+  if !ENV["APP_HOSTNAME"].nil?
```
Comment (scottrobertson) [inline config/environments/development.rb:43]: Could you use '.present?' here instead maybe?

Comment (PromoFaux) [reply config/environments/development.rb:43]: Maybe! I'll have to defer to you as the Ruby expert in this case. I'll give it a go and see if it still works afterward!

I've also got some tweaks to make to the GHA config based on [some advice from @crazy-max](https://github.com/docker/build-push-action/pull/426#issuecomment-932776148)

Comment (PromoFaux) [reply config/environments/development.rb:43]: `.present?` didn't work. I had thought about using `!ENV["APP_HOSTNAME"].nil? && !ENV["APP_HOSTNAME"].empty?`, but I don't know if that's any better

Comment (PromoFaux) [reply config/environments/development.rb:43]: Ah wait, I had the logic reversed... one sec

**Review by scottrobertson — APPROVED** [review]

**Comment by scottrobertson** [comment]
@PromoFaux i merged this, but it broke the Docker stuff. It needs the GHC_TOKEN stuff, as the default token does not have enough permissions to push to the registry.

Can you revert those parts in another PR?

**Comment by PromoFaux** [comment]
Hmmm, that's odd, I made sure it worked before I PR'd - the image pushed to `promofaux/fintech-to-ynab`

I wonder what stops it working here? Will have a play about tomorrow!

**Comment by PromoFaux** [comment]
Ah. I wonder...

What is the `Workflow permissions`  set to in https://github.com/syncforynab/fintech-to-ynab/settings/actions?

On my fork it has defaulted to `Read and write permissions`

**Comment by PromoFaux** [comment]
Looks like it can also be set in the action : https://docs.github.com/en/packages/managing-github-packages-using-github-actions-workflows/publishing-and-installing-a-package-with-github-actions#publishing-a-package-using-an-action

**Pull request merged**